### DISCUSSION TITLE
Trivia bulletin board (#166)

### DISCUSSION THREAD

**Discussion by Farigh** [Farigh/pokeclicker-automation#168] [PR discussions]
The Trivia panel is becoming to big IMO : 
![image](https://user-images.githubusercontent.com/11090416/198347213-42ed5154-d5b6-4b00-8f63-b504e8bf7381.png)

I think it would be better to introduce a new menu mechanic for this one, I'd like it to stay as compact as possible.

One idea would be to add options to display either the full text block, as in your first implementation.
And an option to only add a clickable icon when there is an important info, for example:
  - The are new bulletin board quests available
  - There are uncaught roamers
The only option that should be displayed at any time should be the `Go to` feature

**Comment by Farigh** [comment]
As a first implementation, I would only display `Quest available 📖` if a bultin board is available and has quests in the current region (or subregion) like it's the case for Roamers.

The town info should be added as a tooltip, like roamers info

Clicking on the texte could go to the bulletin board town.

**Reply by Daniferrito** [reply]
I like showing only the text `Quest available 📖`, as that is much more compact. We can add information on WHERE in the tooltip.

I do not think it should only show if the player is in the right region already. The easy to miss quests are the ones in a region you do not usually see. For example, if a new questline appears in Kalos, and the player is in Galar, it will never be seen even with the trivia panel. This is even more common now that they are introducing new questlines (the ones for Lugia and Ho-oh are less than a week old, and I only saw them by chance). Aditionally, if the player is already in the region, they will almost surely see the quest anyways, as the location will be glowing in orange.

I agree that clicking on the text should move the player there (but I dont know if that is an issue cross-region). Probably should happen with the roamers as well (and will try to add it to the PR while I'm at it).

Finally: Do you know how to get the correct text for the name of a region from its ID? I do not find it anywhere.

**Reply by Farigh** [reply]
> I do not think it should only show if the player is in the right region already. The easy to miss quests are the ones in a region you do not usually see. For example, if a new questline appears in Kalos, and the player is in Galar, it will never be seen even with the trivia panel. This is even more common now that they are introducing new questlines (the ones for Lugia and Ho-oh are less than a week old, and I only saw them by chance). Aditionally, if the player is already in the region, they will almost surely see the quest anyways, as the location will be glowing in orange.

You're right, it might be better this way.
In this case, we should not make the text clickable and add the list of available bulletin boards in the tooltip.

> I agree that clicking on the text should move the player there (but I dont know if that is an issue cross-region). Probably should happen with the roamers as well (and will try to add it to the PR while I'm at it).

There is an issue with cross-region moves, however I implemented a check for that:
https://github.com/Farigh/pokeclicker-automation/blob/master/src/lib/Utils/Route.js#L89-L96

However, this should not be done with only a click to the `Quest available 📖` text.
The user should be able to choose where to go, in case of multiple locations.

One idea could be to implement a popup modal that would appear by clicking the text, and this popup would list bulletin board location with a `Go to` button for each.
In case the region is not accessible, the button should be disabled.

> Finally: Do you know how to get the correct text for the name of a region from its ID? I do not find it anywhere.

You could use this : 
https://github.com/pokeclicker/pokeclicker/blob/b7a97950e2436bb47fcfa0fab1013ab338e29b8d/src/modules/subRegion/SubRegions.ts#L37-L39

Calling it like that : 
```javascript
SubRegions.getSubRegionById(regionNumber, subRegionNumber);
```

**Reply by Daniferrito** [reply]
So, to recap:
* Show at all times (only the text `Quest available 📖`)
* Show the list of bulletin boards in the tooltip, showing number of quests, town and subregion `${n} quest(s) available in ${townName} in ${subregionName}`. One line per each town that has missions.
* Do not do anything on click (the modal can come afterwards, but too much hassle for now)

About the region name, I did not realize that some regions have multiple bulletin boards. Kanto has two, and Galar has three. I wanted the name of the region itself, not the subregion, but it is not enough to identify where it is.

**Reply by Farigh** [reply]
That would be it.

> Show at all times  (only the text `Quest available 📖`)

Just to be sure, it should be only be displayed if there is at least one quest

**Comment by Farigh** [comment]
Sorry for the conflict :/

**Reply by Daniferrito** [reply]
I do not appreciate the conflict, but the roamers trivia update is very good.

**Comment by Farigh** [comment]
I lock this conversation, since the PR was merged
